lib/credentials/shared_ini_file_credentials.js

```diff
@@ -48,11 +51,11 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
 
     this.filename = options.filename;
     this.profile = options.profile || process.env.AWS_PROFILE || 'default';
-    this.get(function() {});
```

Comment:
I thought removing `this.get` from the constructor was a safe bet, but there is a scenario that would break existing users with this change. Specifically, if a user is generating presigned url's synchronously, and are not running other operations (a valid use-case), their code would start to fail because the credentials won't have been loaded.

Unfortunately I think it's not safe to remove the extra `get` call currently, but we could in a major version bump in the future.